app/components/ContextMenu/index.tsx

```diff
@@ -50,20 +53,38 @@ export default function ContextMenu({
   const previousVisible = usePrevious(rest.visible);
   const maxHeight = useMenuHeight(rest.visible, rest.unstable_disclosureRef);
   const backgroundRef = React.useRef<HTMLDivElement>(null);
+  const { ui } = useStores();
+  const { t } = useTranslation();
+  const { setOpenCount } = useMenuContext();
 
   React.useEffect(() => {
     if (rest.visible && !previousVisible) {
       if (onOpen) {
         onOpen();
       }
+      if (rest["aria-label"] !== t("Submenu")) {
+        setOpenCount(true);
+      }
```

Comment:
The useEffect doesn't run if I click on MenuItems of Submenu. So it leads to incorrect menuCount.